### PR TITLE
IR-3088-MT-INT-Studio-Link-Component-does-not-redirect-to-the-link

### PR DESCRIPTION
**Pull request by MichaelEstes**
## Summary
Adds a warning message when trying to interact with a scene link in the studio
Adds cleanup functions for intractable component
Fixes race condition with hierarchy panel and click placement when the click placement entity is removed
## References
closes #_insert number here_

### CODE REVIEW

**Review by SamMazerIR — APPROVED** [review]

**Review by SamMazerIR — APPROVED** [review]

File: packages/editor/src/services/EditorServices.ts [hunk 1]
```diff
@@ -86,5 +89,17 @@ export const EditorState = defineState({
     if (!rootEntity) return false
     return !!getState(GLTFModifiedState)[getComponent(rootEntity, SourceComponent)]
   },
-  extension: syncStateWithLocalStorage(['expandedNodes'])
+  extension: syncStateWithLocalStorage(['expandedNodes']),
+  reactor: () => {
+    const linkState = useMutableState(LinkState)
```
Comment (HexaField) [inline packages/editor/src/services/EditorServices.ts:94]: ideally this should not be the responsibility if the EditorState, this should be handled in the link opening logic in client-core

File: packages/editor/src/systems/ClickPlacementSystem.tsx [hunk 2]
```diff
@@ -143,9 +143,9 @@ const ClickPlacementReactor = (props: { parentEntity: Entity }) => {
         (uuid) => uuid !== getComponent(clickState.placementEntity.value, UUIDComponent)
       )
       EditorControlFunctions.removeObject([clickState.placementEntity.value])
-      removeEntity(clickState.placementEntity.value)
       clickState.placementEntity.set(UndefinedEntity)
       SelectionState.updateSelection(selectedEntities)
+      applyIncomingActions()
```
Comment (HexaField) [inline packages/editor/src/systems/ClickPlacementSystem.tsx:148]: Running applyIncomingActions here will likely have unintended consequences, what is the reason for doing this?

File: packages/editor/src/services/EditorServices.ts (hunk 1 above)
Comment (MichaelEstes) [reply packages/editor/src/services/EditorServices.ts:94]: The EngineHooks.tsx file in client-core that handles the scene link logic isn't loaded through the studio flow, I can make it a part of the engine loading flow for the studio though

File: packages/editor/src/systems/ClickPlacementSystem.tsx (hunk 2 above)
Comment (MichaelEstes) [reply packages/editor/src/systems/ClickPlacementSystem.tsx:148]: Ok, I'll try to dig more into the root cause, at the base it seems like there's a race condition with removeObject where there's a frame in the GLTFSnapshotState where the entity no longer exists, but is still listed in the GLTF nodes

**Pull request merged**